### PR TITLE
Run tests with JDK8 in GitHub actions

### PR DESCRIPTION
**Pull request by jbescos**
Adding JDK8 to make sure there are no runtime errors with JDK8.

Currently test are failing because of test errors, which is actually what we wanted:
```
java.lang.ArrayStoreException
	at java.lang.System.arraycopy(Native Method)
	at java.util.ArrayList.toArray(ArrayList.java:414)
	at jakarta.activation.MailcapCommandMap.<init>(MailcapCommandMap.java:181)
	at jakarta.activation.CommandMap.getDefaultCommandMap(CommandMap.java:60)
	at jakarta.activation.DataHandler.getCommandMap(DataHandler.java:128)
	at jakarta.activation.DataHandler.getDataContentHandler(DataHandler.java:590)
	at jakarta.activation.DataHandler.writeTo(DataHandler.java:290)
	at jakarta.mail.internet.MimeUtility.getEncoding(MimeUtility.java:316)
	at jakarta.mail.internet.MimeBodyPart.updateHeaders(MimeBodyPart.java:1580)
	at jakarta.mail.internet.MimeMessage.updateHeaders(MimeMessage.java:2265)
	at jakarta.mail.internet.MimeMessage.saveChanges(MimeMessage.java:2225)
	at jakarta.mail.internet.MimeMessage.writeTo(MimeMessage.java:1904)
	at com.sun.mail.smtp.SMTPTransport.sendMessage(SMTPTransport.java:1329)
```

This will work when this [PR](https://github.com/eclipse-ee4j/angus-activation/pull/11) is merged and we update the version of angus-activation.

I keep this as a draft till that moment happens to keep the builds green.

You can test this following these steps: https://github.com/eclipse-ee4j/angus-activation/pull/11

### CODE REVIEW

**Comment by lukasj** [comment]
time to rebase this on master and finalize...

**Comment by jbescos** [comment]
> time to rebase this on master and finalize...

Done, but we have to wait for a new release of activation. Then I will update the pom.xml to use it, and we can merge it.

**Comment by lukasj** [comment]
ang-activation 2.0.0 is already in staging and integrated here, just rebase on top of master...

**Comment by jbescos** [comment]
> ang-activation 2.0.0 is already in staging and integrated here, just rebase on top of master...

It looks there is a new issue:
Error:  COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
Error:  /home/runner/work/angus-mail/angus-mail/core/src/main/java/com/sun/mail/nativeimage/AngusMailFeature.java:[23,38] cannot access org.graalvm.nativeimage.hosted.Feature
  bad class file: /home/runner/.m2/repository/org/graalvm/sdk/graal-sdk/22.3.0/graal-sdk-22.3.0.jar(org/graalvm/nativeimage/hosted/Feature.class)
    class file has wrong version 55.0, should be 52.0
    Please remove or make sure it appears in the correct subdirectory of the classpath.

It seems that graal dependency was not compiled for JDK 8.

File: pom.xml [hunk 1]
```diff
@@ -91,7 +91,7 @@ Copyright &#169; 2019, ${current.year} Eclipse Foundation. All rights reserved.]
         <servlet-api.version>5.0.0</servlet-api.version>
         <angus.activation.version>2.0.0</angus.activation.version>
         <angus-mail.version>${project.version}</angus-mail.version>
-        <graal.sdk.version>22.3.0</graal.sdk.version>
+        <graal.sdk.version>22.0.0</graal.sdk.version>
```
Comment (lukasj) [inline pom.xml:94]: prefer exclusion of the file from the build on JDK 8 over changing graalvm version

**Review by lukasj — APPROVED** [review]
LGTM

**Pull request merged**